Adds README.md (Markdown) for enhanced formatting beyond README (the one without the extension)

andyatkinson opened the pull request:
Adds README.md (Markdown) which supplements the README file without an extension.

For places like GitHub and GitLab that read Markdown files, formatting is applied for:

* Section headings
* Links

samay-sharma commented on README.md:
```diff
@@ -0,0 +1,25 @@
+# PostgreSQL
```
You should rebase. Tom committed https://github.com/postgres/postgres/commit/e2b73f4a4de6c2df6a1e623ad06b42f1bb2471ad which makes some of the same changes as you.

I'd also include the links which he has in his PR (eg. add a link to docs somewhere).

andyatkinson replied:
"Installation from Source Code" docs link was added.

samay-sharma commented on README.md:
```diff
@@ -0,0 +1,25 @@
+transactions, foreign keys, subqueries, triggers, user-defined types
+and functions. This distribution also contains C language bindings.
+
+The latest version of this software, and related software, may be
```
Can merge this and the below section and include links to download binaries, instructions to build from source, link to website and docs.

andyatkinson committed:
Adds README.md, intended to replace README

Goals: README.md is intended to replace README and provide the following
benefits

* Helpful section headings
* Useful links for further reading
* More attractive presentation from Markdown-formatted plain text

andyatkinson committed:
Updates based on feedback

Merge upstream changes into branch

andyatkinson committed:
Adjustments

andyatkinson committed:
Mention GitLab

GitLab mirror[1] also presents Markdown

[1]: https://gitlab.com/postgres/postgres

andyatkinson committed:
Merge branch 'master' into chore/enhanced-main-readme

andyatkinson committed:
Consolidate sections

Incorporate feedback from Samay
